plutus-contract/src/Plutus/Contract/StateMachine.hs

```diff
@@ -462,7 +462,7 @@ mkStep client@StateMachineClient{scInstance} input = do
         Nothing -> pure $ Left $ InvalidTransition Nothing input
         Just (onChainState, utxo) -> do
             let (TypedScriptTxOut{tyTxOutData=currentState, tyTxOutTxOut}, txOutRef) = onChainState
-                oldState = State{stateData = currentState, stateValue = Ledger.txOutValue tyTxOutTxOut}
+                oldState = State{stateData = currentState, stateValue = Ledger.txOutValue tyTxOutTxOut <> inv (SM.threadTokenValueOrZero scInstance)}
```

Comment:
Both of these are worth a comment!

Reply:
Would it be simpler to rather only add the token value to the current value in the transition where we mint it?

Reply:
We need to keep adding it because the transition can just ignore the old value. In fact all our tests did this, that’s why this bug did not surface earlier.

Reply:
I see. Maybe we really want something like `(\/) :: Value -> Value -> Value` so we could write `newValue = transitionValue \/ threadTokenValue` or something.

Reply:
What am I talking about, we literally have that.

Reply:
Okay, so I suggest we do that: that lets us always add it in without duplicating.

Reply:
I think it is actually nice to hide the thread token value from the client code? I haven't seen examples where the oldValue is used, but I guess if you want to do something with that it would be convenient if you don't have to take the thread token value into account.

Reply:
Oh I see, so the idea is that the transition function never sees the thread token at all. That's a good point, I guess. I'm not sure what is nicer.